api/src/main/java/org/pmiops/workbench/access/AccessSyncServiceImpl.java

```diff
@@ -84,6 +94,20 @@ public DbUser updateUserAccessTiers(DbUser dbUser, Agent agent) {
     return userDao.save(dbUser);
   }
 
+  private void createVwbUserIfNeeded(
+      DbUser dbUser, List<DbAccessTier> previousAccessTiers, List<DbAccessTier> newAccessTiers) {
+    // This means that the user has been granted access to a tier for the first time. Then perform
+    // the VWB creation logic.
+    if (previousAccessTiers.isEmpty() && !newAccessTiers.isEmpty()) {
```

Comment:
can we extract this to be a common method, the part to check if user first get tier access?